src/files/glob-source.js

```diff
@@ -0,0 +1,89 @@
+'use strict'
+
+const fs = require('fs-extra')
+const glob = require('it-glob')
+const Path = require('path')
+const errCode = require('err-code')
+
+/**
+* Create an async iterator that yields paths that match requested file paths.
+*
+* @param {String} ...paths File system path(s) to glob from
+* @param {Object} [options] Optional options
+* @param {Boolean} [options.recursive] Recursively glob all paths in directories
+* @param {Boolean} [options.hidden] Include .dot files in matched paths
+* @param {Array<String>} [options.ignore] Glob paths to ignore
+* @param {Boolean} [options.followSymlinks] follow symlinks
+* @yields {Object} File objects in the form `{ path: String, content: AsyncIterator<Buffer> }`
+*/
+module.exports = async function * globSource (...args) {
+  const options = typeof args[args.length - 1] === 'string' ? {} : args.pop()
+  const paths = args
+
+  const globSourceOptions = {
+    recursive: options.recursive,
+    glob: {
+      dot: Boolean(options.hidden),
+      ignore: Array.isArray(options.ignore) ? options.ignore : [],
+      follow: options.followSymlinks != null ? options.followSymlinks : true
+    }
+  }
+
+  // Check the input paths comply with options.recursive and convert to glob sources
+  for (const path of paths) {
+    if (typeof path !== 'string') {
+      throw errCode(
+        new Error(`Path must be a string`),
+        'ERR_INVALID_PATH',
+        { path }
+      )
+    }
+
+    const absolutePath = Path.resolve(process.cwd(), path)
+    const stat = await fs.stat(absolutePath)
+    const prefix = Path.dirname(absolutePath)
+
+    for await (const entry of toGlobSource({ path, type: stat.isDirectory() ? 'dir' : 'file', prefix }, globSourceOptions)) {
+      yield entry
+    }
+  }
+}
+
+async function * toGlobSource ({ path, type, prefix }, options) {
+  options = options || {}
+
+  const baseName = Path.basename(path)
+
+  if (type === 'file') {
+    yield {
+      path: baseName.replace(prefix, ''),
```

Comment:
I had to make changes here to get tests to pass https://github.com/ipfs/js-ipfs-http-client/pull/1087/files#diff-b1b66f6cd58278b2931897df7251ad19

Reply:
Solved: My version is passing a different `absolute` option value. This *should* be fine!